lib/jsdom/browser/Window.js

```diff
@@ -403,6 +404,21 @@ function matchesDontThrow(el, selector) {
 }
 
 function startTimer(window, startFn, stopFn, callback, ms) {
+  if (typeof callback !== "function") {
+    const code = String(callback);
+    callback = window._globalProxy.eval.bind(window,
+      `(function(){` + code + `}())\n//# sourceURL=${window.location.href}`);
```

Comment:
Actually works, it's a v8 thing. vm is ~5x slower than eval, see https://github.com/facebook/jest/pull/635#issuecomment-162708673.